internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -21,54 +33,242 @@ var (
 	serverKey []byte
 )
 
+// TODO(rmehta19): Consider refactoring stream creation for client/server into
+// a helper function.
+
 // GetTlsConfigurationForClient returns a tls.Config instance for use by a client application.
 func GetTlsConfigurationForClient(serverHostname string) *tls.Config {
```

Comment:
I think the stream need to be passed to GetTlsConfigurationForClient rather than creating the stream inside the implementation.

Reply:
Thanks for pointing this out! I moved the setup of the bidirectional stream to the client and server handshake, which call GetTlsConfigurationForClient/Server, and modified the tls config store unit store accordingly